include/multi_blas_helper.cuh

```diff
@@ -95,18 +166,20 @@ namespace quda
       using SpinorW = SpinorX;
 
       // compute the size remaining for the Y and W accessors
-      constexpr int arg_size = (MAX_ARG_SIZE - sizeof(int)                                    // NYW parameter
-                                - sizeof(SpinorX[NXZ])                                        // SpinorX array
-                                - (Functor::use_z ? sizeof(SpinorZ[NXZ]) : sizeof(SpinorZ *)) // SpinorZ array
-                                - sizeof(Functor)                                             // functor
-                                - sizeof(int)                                                 // length parameter
-                                - (!Functor::use_w ? sizeof(SpinorW *) : 0)   // subtract pointer if not using W
-                                - (Functor::reducer ? sizeof(ReduceArg<device_reduce_t>) : 0) // reduction buffers
-                                - 16) // there seems to be 16 bytes other argument space we need
+      constexpr auto arg_size = (max_arg_size<Functor>()
+                                 - sizeof(kernel_param<>)                                      // kernel_param parent
+                                 - sizeof(int)                                                 // NYW parameter
+                                 - sizeof(SpinorX[NXZ])                                        // SpinorX array
+                                 - (Functor::use_z ? sizeof(SpinorZ[NXZ]) : sizeof(SpinorZ *)) // SpinorZ array
+                                 - sizeof(Functor)                                             // functor
+                                 - sizeof(dim3)                                                // threads parameter
+                                 - (!Functor::use_w ? sizeof(SpinorW *) : 0)                   // subtract pointer if not using W
+                                 - (Functor::reducer ? sizeof(ReduceArg<device_reduce_t>) : 0) // reduction buffers
+                                 - 12) // extra 12 bytes subtracted for alignment roundup
```

Comment:
Is there a way to make this a little more robust (less of a "magic number")? More specifically---if we know the alignment (at compile time), can we explicitly round up/down to the known alignment requirement?

Reply:
Yes, likely I can do something here.  I'll stew on this overnight and apply something tomorrow.

Reply:
Ok, I've realized that this correction term is no longer needed due to the rewrite of the classes in this PR.  I've remove this  term, and verified that all is working fine now.  Checked against both nvcc and nvc++, running blas_test for all 128^2 combinations of `--nsrc` and `--msrc` between 1 and 128.  Noting that it was this testing that led to me finding #1187.